scripts/import_osm.py

```diff
@@ -0,0 +1,111 @@
+import osmnx as ox
+from ipygis import get_connection_url
+from sqlalchemy import create_engine
+from sqlalchemy.engine.base import Engine
+from sqlalchemy.schema import CreateSchema
+from geoalchemy2 import Geometry, WKTElement
+from typing import Dict
+
+# test simple import now, convert to module later
+sys.path.insert(0, "..")
+from models import OSMPoint
+from osm_tags import tags_to_filter
+
+
+class OsmImporter(object):
+    """Import OSM data to PostGIS using Overpass API."""
+
+    def __init__(self, args: Dict):
+        args = self._parse_args(args)
+        self.slug = args["slug"]
+        self.bbox = args["bbox"]
+        self.city = args["city_name"]
+        self._engine = self._create_engine()
+
+    @staticmethod
+    def _parse_args(args) -> Dict:
+        """Parse input arguments to a dict."""
+        slug = args["city"]
+        if not slug:
+            raise AssertionError("You must specify the city name.")
+        # BBOX (minx, miny, maxx, maxy) takes precedence over city
+        arg_bbox = args["bbox"]
+        if arg_bbox:
+            minx, miny, maxx, maxy = map(float, arg_bbox.split(", "))
+            city_name = ""
+        else:
+            # support old city parameter too, if we want to let osmnx do geocoding
+            if slug == "COP":
+                city_name = "Copenhagen, Denmark"
+                minx, miny, maxx, maxy = 12.42000, 55.61000, 12.65000, 55.78000
+            elif slug == "HEL":
+                city_name = "Helsinki, Finland"
+                minx, miny, maxx, maxy = 24.82345, 60.14084, 25.06404, 60.29496
+            elif slug == "WAR":
+                city_name = "Warsaw, Poland"
+                minx, miny, maxx, maxy = 20.79057, 52.09901, 21.31300, 52.38502
+            else:
+                raise AssertionError("You must specify a bounding box or a known city.")
+
+        return {"slug": slug, "bbox": (minx, miny, maxx, maxy), "city_name": city_name}
+
+    def _create_engine(self) -> Engine:
+        """Sets up and returns a DB engine."""
+        sql_url = get_connection_url(dbname="geoviz")
+        engine = create_engine(sql_url)
+        return engine.execution_options(
+            schema_translate_map={"schema": self.slug}
+        )
+
+    def run(self):
+        self._initialise_db()
+        print("Fetching OSM data from overpass API...")
+        pois = self._get_amenities()
+        pois = pois.to_crs(epsg=3035)
+        pois.geometry = pois.centroid
+        pois = pois.to_crs(epsg=4326)
+
+        pois = pois.reset_index()
+        tag_columns = list(pois.columns)
+        tag_columns.remove("osmid")
+        tag_columns.remove("geometry")
+
+        pois = pois.rename(columns={"osmid": "node_id"})
+        pois = pois.rename(columns={"geometry": "geom"})
+        pois = pois.set_index("node_id")
+        pois["geom"] = pois["geom"].apply(lambda geom: WKTElement(geom.wkt, srid=4326))
+
+        pois["tags"] = pois[tag_columns].apply(lambda x: x.to_json(), axis=1)
```

Comment:
This one results in a huge json object of tags with all possible json values. I don't think we want all these strings with nulls in the database, it becomes huge and unreadable 😢 
```
{"bar": null, "dog": null, "ele": null, "fax": null, "fee": null, "gay": null, "hgv": null, "lit": null, "mtb": null, "pub": null, "ref": null, "url": null, "area": null, "beer": null, "cafe":
 null, "club": null, "cold": null, "date": null, "dogs": null, "fish": null, "food": null, "foot": null, "heat": null, "male": null, "meat": null, "name": "Polyteekkarimuseo", "note": null, "roof": null, 
"room": null, "ship": null, "shop": null, "type": null, "ways": null, "wifi": null, "FIXME": null, "asian": null, "blind": null, "books": null, "brand": null, "craft": null, "depth": null, "email": null, 
"fixme": null, "floor": null, "games": null, "halal": null, "hoops": null, "horse": null, "image": null, "is_in": null, "layer": null, "level": null, "lgbtq": null, "lunch": null, "music": null, "nodes": 
null, "parts": null, "phone": null, "place": null, "power": null, "sauna": null, "sells": null, "shoes": null, "sport": null, "stars": null, "table": null, "taxon": null, "topic": null, "trade": null, "ac
cess": null, "bakery": null, "beauty": null, "boules": null, "branch": null, "brunch": null, "charge": null, "colour": null, "erotic": null, "female": null, "fenced": null, "hazard": null, "height": null,
 "indoor": null, "length": null, "mobile": null, "museum": null, "noname": null, "office": null, "opened": null, "origin": null, "public": null, "rental": null, "repair": null, "robbed": null, "screen": n
ull, "shower": null, "source": null, "toilet": null, "unisex": null, "CHECKME": null, "address": null, "alcohol": null, "amenity": null, "archery": null, "barrier": null, "bicycle": null, "brewery": null,
 "butcher": null, "chicken": null, "clothes": null, "covered": null, "cuisine": null, "culture": null, "date_on": null, "disused": null, "ele:msl": null, "highway": null, "karaoke": null, "landuse": null,
 "leisure": null, "massage": null, "max_age": null, "min_age": null, "name:ar": null, "name:ca": null, "name:da": null, "name:de": null, "name:en": null, "name:es": null, "name:et": null, "name:fa": null,
 "name:fi": null, "name:fr": null, "name:id": null, "name:it": null, "name:ja": null, "name:ko": null, "name:nl": null, "name:nn": null, "name:no": null, "name:ro": null, "name:ru": null, "name:se": null,
 "name:sv": null, "name:uk": null, "name:zh": null, "natural": null, "organic": null, "outdoor": null, "parking": null, "phone_1": null, "portier": null, "rooftop": null, "sensory": null, "service": null,
 "shelter": null, "smoking": null, "surface": null, "terrace": null, "toilets": null, "tourism": "museum", "twitter": null, "vending": null, "website": "https://www.polyteekkarimuseo.fi/", "alt_name": nul
l, "boundary": null, "building": null, "capacity": null, "currency": null, "delivery": null, "diet:raw": null, "drive_in": null, "end_date": null, "entrance": null, "facebook": null, "fuel:cng": null, "fu
el:e10": null, "fuel:e85": null, "fuel:lng": null, "fuel:lpg": null, "gasoline": null, "historic": null, "identity": null, "int_name": "Polytechnical Museum", "loc_name": null, "location": null, "man_made": null, "material": null, "memorial": null, "motorcar": null, "name:old": null, "old_name": null, "operator": null, "provides": null, "religion": null, "seasonal": null, "shooting": null, "takeaway": null, "veikkaus": null, "was:name": null, "was:shop": null, "waterway": null, "wikidata": "Q11888594", "addr:city": "Espoo", "addr:door": null, "addr:full": null, "addr:name": null, "addr:unit": null, "amenity_1": null, "architect": null, "athletics": null, "automated": null, "bath:type": null, "breakfast": null, "cinema:3D": null, "cocktails": null, "cuisine_1": null, "cuisine_2": null, "diet:meat": null, "ele:wgs84": null, "fee:adult": null, "fee:child": null, "fireplace": null, "fuel:MPÖ": null, "full_name": null, "furniture": null, "glass:new": null, "high_seat": null, "highchair": null, "ice_cream": null, "leaf_type": null, "leisure_1": null, "max_level": null, "min_level": null, "ref:vatin": null, "ship:type": null, "substance": null, "wholesale": null, "wikipedia": "fi:Polyteekkarimuseo", "access:dog": null, "addr:floor": null, "addr:place": null, "beer_brand": null, "car_repair": null, "created_by": null, "diet:halal": null, "diet:vegan": null, "dispensing": null, "drink:wine": null, "ele:source": null, "embankment": null, "fair_trade": null, "fee:family": null, "fence_type": null, "healthcare": null, "high_chair": null, "lunch:menu": null, "lunch:warm": null, "maintained": null, "piste:type": null, "playground": null, "restaurant": null, "roof:shape": null, "sauna:fuel": null, "short_name": null, "start_date": null,
```

Reply:
I think at least filter out *all* the null values.

Or, even better, since we are only interested in amenities, import `tags_to_filter` from `osm_tags.py` and *only* save those, and filter out all the nulls.

Otherwise, the tag filter doesn't work correctly I'm afraid, since it checks the presence of OSM tags, and to_json() seems to add
*every* tag to *every* object.

Reply:
Will fix this later once I get the testing environment to work again...

Reply:
Should be fixed now.